src/routes/homework/CreateHomework.tsx

```diff
@@ -0,0 +1,492 @@
+            {question.type}
+          </span>
+        </div>
+        <span
+          className={styles['question-title']}
+          title={question.title}
+        >
+          {question.title}
+        </span>
+      </div>
+      <button
+        type="button"
+        className={styles['question-delete']}
+        onClick={() => onDelete(question)}
+      >
+        <IconDelete />
+      </button>
+    </Card>
+  );
+}
+
+const INITIAL_INPUT_ERRORS = {
+  class_id: '',
+  difficulty_id: '',
+  title: '',
+  open_questions: '',
+  closed_questions: '',
+  deadline: '',
+  questions_ids: '',
+};
+
+type InputErrors = typeof INITIAL_INPUT_ERRORS;
+
+type UpdateInputError = {
+  type: 'update';
+  payload: { [Key in keyof InputErrors]?: string };
+};
+type ResetInputErrors = { type: 'reset'; payload: InputErrors };
+type InputErrorActions = UpdateInputError | ResetInputErrors;
+
+function inputErrorsReducer(state: InputErrors, action: InputErrorActions) {
+  switch (action.type) {
+    case 'update':
+      return { ...state, ...action.payload };
+    case 'reset':
+      return { ...action.payload };
+    default:
+      return state;
+  }
+}
+
+const INITIAL_HOMEWORK = (
+  class_id: ItemList | undefined,
+  difficulty: 1 | 2 | 3
+): HomeworkRequest => ({
+  class_id: class_id || '',
+  difficulty_id: difficulty,
+  title: '',
+  open_questions: undefined,
+  closed_questions: undefined,
+  deadline: '',
+  questions_ids: [...questionList],
+});
+
+type UpdateClass = { type: 'class'; payload: string | ItemList };
+type UpdateDifficulty = { type: 'difficulty'; payload: 1 | 2 | 3 };
+type UpdateTitle = { type: 'title'; payload: string };
+type UpdateOpenQuestions = { type: 'open_questions'; payload: number };
+type UpdateClosedQuestions = { type: 'closed_questions'; payload: number };
+type UpdateDeadline = { type: 'deadline'; payload: string };
+type UpdateQuestions = { type: 'questions'; payload: Question[] };
+type ResetHomework = { type: 'reset'; payload: HomeworkRequest };
+type ReducerActions =
+  | UpdateClass
+  | UpdateDifficulty
+  | UpdateTitle
+  | UpdateOpenQuestions
+  | UpdateClosedQuestions
+  | UpdateDeadline
+  | UpdateQuestions
+  | ResetHomework;
+
+function homeworkRequestReducer(
+  state: HomeworkRequest,
+  action: ReducerActions
+): HomeworkRequest {
+  switch (action.type) {
+    case 'class':
+      return { ...state, class_id: action.payload };
+
+    case 'title':
+      return { ...state, title: action.payload };
+
+    case 'open_questions':
+      return { ...state, open_questions: action.payload };
+
+    case 'closed_questions':
+      return { ...state, closed_questions: action.payload };
+
+    case 'deadline':
+      return { ...state, deadline: action.payload };
+
+    case 'questions':
+      return { ...state, questions_ids: action.payload };
+
+    case 'reset':
+      return { ...action.payload };
+
+    default:
+      return state;
+  }
+}
+
+type Props = {
```

Comment:
Probably this should be a type since I find myself using it a lot, and incoordination is likely to happen between files if a change is required